miden-lib/asm/miden/kernels/tx/tx.masm

```diff
@@ -111,7 +111,7 @@ export.create_note
     assertz.err=ERR_NOTE_INVALID_TAG_HIGH_BIT_SET
     # => [tag_low, note_type, ASSET, tag, note_type, RECIPIENT]
 
-    u32shr.30 u32and assertz.err=ERR_NOTE_INVALID_TAG_PREFIX_FOR_TYPE
```

Comment:
This was a bug, the `u32shr.30` gets the two highest bits of the tag. That should be asserted to be equal to the note's type.

Reply:
> I don't think we can enforce that the first two bits of the tag are equal to the note type

Why is that?

> The rules are as follows

Seems the rule is that the note type is a bitmaks, and the corresponding bits must be zero?

Reply:
> > I don't think we can enforce that the first two bits of the tag are equal to the note type
> 
> Why is that?

Note type by itself is not enough to capture all the data we want to convey (specifically, it is missing the part about who is the intended recipient of the note - i.e. a single account or any account).

> Seems the rule is that the note type is a bitmaks, and the corresponding bits must be zero?

Given the note tag formats described in https://github.com/0xPolygonMiden/miden-base/issues/402#issuecomment-2008740524, that's how it works out to be (though, I didn't plan for it to be like that originally).

Reply:
> Note type by itself is not enough to capture all the data we want to convey (specifically, it is missing the part about who is the intended recipient of the note - i.e. a single account or any account).

Yes, I wasn't proposing removing the address from the tag. The question was why the note type can't be encoded directly to it.

Reply:
> Given the note tag formats described in https://github.com/0xPolygonMiden/miden-base/issues/402#issuecomment-2008740524, that's how it works out to be (though, I didn't plan for it to be like that originally).

So that was mostly by accident?